src/bdev.rs

```diff
@@ -303,8 +310,47 @@ where
         }
     }
 
+    /// Returns true if this Bdev supports the ZNS command set.
+    pub fn is_zoned(&self) -> bool {
+        unsafe { spdk_bdev_is_zoned(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get device zone size in logical blocks.
+    pub fn zone_size(&self) -> u64 {
+        unsafe { spdk_bdev_get_zone_size(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get the number of zones for the given device.
+    pub fn num_zones(&self) -> u64 {
+        unsafe { spdk_bdev_get_num_zones(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get device maximum zone append data transfer size in logical blocks.
+    pub fn max_zone_append_size(&self) -> u32 {
+        unsafe { spdk_bdev_get_max_zone_append_size(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get device maximum number of open zones.
+    pub fn max_open_zones(&self) -> u32 {
+        unsafe { spdk_bdev_get_max_open_zones(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get device maximum number of active zones.
+    pub fn max_active_zones(&self) -> u32 {
+        unsafe { spdk_bdev_get_max_active_zones(self.unsafe_inner_ptr()) }
+    }
+
+    /// Get device optimal number of open zones.
+    pub fn optimal_open_zones(&self) -> u32 {
+        unsafe { spdk_bdev_get_optimal_open_zones(self.unsafe_inner_ptr()) }
+    }
+
     /// Determines whenever the Bdev supports the requested I/O type.
     pub fn io_type_supported(&self, io_type: IoType) -> bool {
+        if self.is_zoned() && io_type == IoType::ZoneAppend {
+            debug!("Always claiming to support zone append such that the exposed NVMe-oF nexus is not set to read only by the kernel. The nexus logic strictly rejects zone append commands.");
```

Comment:
Should this perhaps be just a comment? Doesn't carry any information pertaining to a specific device and it might flood the logs if there's a lot of io_supported checks.

Reply:
Agree :)